v0/destinations/heap/transform.js

```diff
@@ -39,20 +83,32 @@ function commonPayload(message, rawPayload, type) {
   const propertiesObj = {};
   let propsArray;
   let rudderPropertiesObj;
+  let email;
   switch (type) {
     case EventType.TRACK:
       propsArray = get(message, "properties")
         ? Object.keys(message.properties)
         : null;
       rudderPropertiesObj = message.properties;
-      rawPayload.identity = message.context.traits.email;
+      email =
+        (message.context.traits ? email : null) || message.properties.email;
```

Comment:
(message.context.traits ? email : null)  ==> email is undefined here right?
so rawPayload.identity = null is valid but rawPayload.identity = undefined (previous version) is not?

Reply:
changed the logic